crates/sui-light-client/src/main.rs

```diff
@@ -93,12 +116,42 @@ struct Config {
 
     //  Genesis file name
     genesis_filename: PathBuf,
+
+    /// Object store url
+    object_store_url: String,
+
+    /// GraphQL endpoint
+    graphql_url: String,
 }
 
-impl Config {
-    pub fn rest_url(&self) -> String {
-        format!("{}/rest", self.full_node_url)
-    }
+async fn query_last_checkpoint_of_epoch(config: &Config, epoch_id: u64) -> anyhow::Result<u64> {
+    // GraphQL query to get the last checkpoint of an epoch
+    let query = json!({
+        "query": "query ($epochID: Int) { epoch(id: $epochID) { epochId checkpoints(last: 1) { nodes { sequenceNumber } } } }",
```

Comment:
Do you need the `epochId` in the query result?
```suggestion
        "query": "query ($epochID: Int) { epoch(id: $epochID) { checkpoints(last: 1) { nodes { sequenceNumber } } } }",
```

Reply:
Good catch, thanks!